src/sentry/options/manager.py

```diff
@@ -369,3 +450,47 @@ def get_last_update_channel(self, key: str) -> Optional[UpdateChannel]:
         # be applied evaluating all the possible drift cases.
         opt = self.lookup_key(key)
         return self.store.get_last_update_channel(opt)
+
+    def can_update(self, key: str, value, channel: UpdateChannel) -> Optional[NotWritableReason]:
+        """
+        Return the reason the provided channel cannot update the option
+        to the provided value or None if there is no reason and the update
+        is allowed.
+        """
+
+        required_flag = WRITE_REQUIRED_FLAGS.get(channel)
+        opt = self.lookup_key(key)
+        if opt.flags & (FLAG_NOSTORE | FLAG_IMMUTABLE):
+            return NotWritableReason.READONLY
+        if opt.flags & FLAG_PRIORITIZE_DISK and settings.SENTRY_OPTIONS.get(key):
+            return NotWritableReason.OPTION_ON_DISK
```

Comment:
Some version of this comment might be helpful here.  It could be 10 years before someone touches this again.  I also think it could be repeated on the `NotWritableReason` enum.  The language can vary given the context of each.

> PRIORITIZE_DISK means: trust the settings and prevent it from being stored in the DB IF the option is actually set in the settings file.
If it is not set in the settings file, then the option can be remotely changed.